### PR TITLE
Preserve case in citekey rather than slugging it

### LINKED ISSUE #16: Preserve case when creating file name for notes
I saw that the notes file created by `org-roam-bibtex-edit-notes` uses the lower-case version of the bibtex key + `.org`.  Preserving the case might be better because that would allow helm/ivy-bibtex to find the notes by simply setting `bibtex-completion-notes-path` to `org-roam-directory`.

### PR DESCRIPTION
**Pull request by zaeph**
Fixes #16.

@myshevchuk Can you think of any problem that'd come with that change?

### CODE REVIEW

File: org-roam-bibtex.el [hunk 1]
```diff
@@ -490,7 +490,7 @@ calling any Org-roam functions."
               (let ((org-roam-capture--context 'ref)
                     (org-roam-capture--info (list (cons 'title title)
                                                   (cons 'ref citekey-formatted)
-                                                  (cons 'slug (org-roam--title-to-slug citekey)))))
```
Comment (myshevchuk) [inline org-roam-bibtex.el:493]: Please, revert this change.

**Comment by myshevchuk** [comment]
@zaeph I don't see any problems here. 

I'd suggest, however, to keep `(cons 'slug (org-roam--title-to-slug citekey)` or even better set it to `(cons 'slug (org-roam--title-to-slug citekey)` since it's irrelevant for our templates now. The reason is that `slug` is used internally by Org-roam as a kind of unique key for the database, which is all I know so far. But I can anticipate many unknowns like the exact details of how and where `slug` is used in Org-roam and how its use there will change in future. Keep in mind, we are actually hacking into Org-roam by using its internal `org-roam--` functions and such internals may! change in future without any notice, since they are not part of the API.

File: org-roam-bibtex.el (hunk 1 above)
Comment (zaeph) [reply org-roam-bibtex.el]: Reverted in 4df39a7cddc2e7ba19154fa24db39f9fe3349292.

**Comment by zaeph** [comment]
Do we want to make `(("citekey" . "=key="))` the new default for `org-roam-bibtex-preformat-keywords`?  That'd allow us to use `citekey` instead of `=key=` in the template, which might be a little more obvious.

**Comment by zaeph** [comment]
@myshevchuk All good in a81106d3b74f274f5dc06cccfb832c86a379832b?

**Comment by myshevchuk** [comment]
Yes, why not? I think at this stage we can push our changes without like consulting each other.

Could you please add a notice to the Readme that Org-roam-bibtex is alpha and unstable until the first release, and breaking changes are going to happen? Once we've reached version 1.0, we'll need to freeze names of public functions and variables and their arguments, and indeed exercise care in changing even minor defaults.

**Comment by zaeph** [comment]
> Yes, why not? I think at this stage we can push our changes without like consulting each other.

Sure, sorry for pinging you on this one.  With the little fuck-up I had last time, I felt it was warranted to get your opinion on changes, especially those that include changing the default template.  Jumping back and forth between Org-roam and org-roam-bibtex dev, I'm forgetting some of the features we've implemented, and that's why I wanted your opinion.  If you're on IRC or on Slack, maybe we could just discuss those points over there to avoid cluttering each other's inboxes and notifications? :)

> Could you please add a notice to the Readme that Org-roam-bibtex is alpha and unstable until the first release, and breaking changes are going to happen? Once we've reached version 1.0, we'll need to freeze names of public functions and variables and their arguments, and indeed exercise care in changing even minor defaults.

Sure thing, I'll do that.  Thanks for the suggestion!

**Comment by myshevchuk** [comment]
@zaeph take it easy, no need to feel sorry. Feel free to experiment, things are not irreversible after all 😉 I'm not on IRC nor Slack, so don't hesitate to tag me here. It's just that I don't always have time to respond immediately after getting the message. And I don't want you to wait for too long with bug-fix or smaller improvement changes.

**Comment by zaeph** [comment]
> @zaeph take it easy, no need to feel sorry. Feel free to experiment, things are not irreversible after all wink I'm not on IRC nor Slack, so don't hesitate to tag me here. It's just that I don't always have time to respond immediately after getting the message. And I don't want you to wait for too long with bug-fix or smaller improvement changes.

Sure thing, and thanks for the help since the beginning.  I might have worked on some UX elements, but you did the bulk of the work, most motably writing the template expansion and all that.

**Comment by myshevchuk** [comment]
You did a really great job of making and documenting the package and pushing it to MELPA. And the design ideas are as important to the success as is their implementation. I'd say we make a good team.  I'm not a professional programmer either, just have been using Emacs for maybe 10 years now, so have a little bit more experience.

**Comment by zaeph** [comment]
> You did a really great job of making and documenting the package and pushing it to MELPA. And the design ideas are as important to the success as is their implementation. I'd say we make a good team. I'm not a professional programmer either, just have been using Emacs for maybe 10 years now, so have a little bit more experience.

Hehe, you've got me beat my 3 years, so hopefully I've got that to look forward to.

Thanks for the kind word.  Let's keep up the good work!
